File: backend/Chef.DAL/Entities/ComponentIngredient.cs
using Chef.DAL.Entities.Common;

namespace Chef.DAL.Entities
{
    public class ComponentIngredient : Entity
    {
        public int RecipeComponentId { get; set; }
        public int IngredientId { get; set; }
        public double Quantity { get; set; }
        public string Description { get; set; }
        public Ingredient Ingredient { get; set; }
        public RecipeComponent RecipeComponent { get; set; }
        public int? VolumeMetricId { get; set; }
        public VolumeMetric VolumeMetric { get; set; }
    }
}

File: frontend/src/app/shared/models/component-ingredient/component-ingredient.ts
import { Category } from "../category/category";
import { Ingredient } from "../ingredient/ingredient";
import { VolumeMetric } from "../volume-metric/volume-metric";

export interface ComponentIngredient {
    id: number;
    category: Category;
    ingredient: Ingredient;
    quantity: number;
    description: string;
    volumeMetric: VolumeMetric;
}